src/diskquota_utility.c

```diff
@@ -1709,3 +1708,51 @@ check_hash_fullness(HTAB *hashp, int max_size, const char *warning_message, Time
 
 	return HASH_FIND;
 }
+
+int
+SPI_connect_wrapper(void)
+{
+	int rc;
+	int state = 0;
+
+	SetCurrentStatementStartTimestamp();
+
+	if (!IsTransactionState())
+	{
+		StartTransactionCommand();
+		state |= IS_UNDER_TRANSACTION;
+
+		PushActiveSnapshot(GetTransactionSnapshot());
+		state |= IS_ACTIVE_SNAPSHOT_PUSHED;
```

Comment:
Can we change state one time: `state |= IS_ACTIVE_SNAPSHOT_PUSHED | IS_UNDER_TRANSACTION;`?

Reply:
No, because the `PushActiveSnapshot(GetTransactionSnapshot());` command may fail.